### DISCUSSION TITLE
Rond Rows Filter with $in operator

### DISCUSSION THREAD

**Discussion by SimoneBergonzi** [mia-platform/community#268] [Q&A] [answered]
### Which section?

Console

### Which is the version?

v9.5.3

### Description

Hi,

Is it possible to generate a mongoDB filter using the operator `$in` configuring a Row Filter with RBAC?

This is our use case:
The users have a list of companies on which they can operate:
```
{
  "metadata": {
    "managed_companies": ["company1", "company2"]
  }
}
```

When performing a GET on `/companies/`, we would like to filter on the companies returning only the ones managed by the authenticated user, that is:
```
"id": {
  $in: managed_companies
}
```

where `managed_companies` is the array of companies retrieved from the user metadata.

In the documentation I can find examples only with operators
- **=** (-> `$eq`)
- **>=** (-> `$gte`)
- **<=** (-> `$lte`)

Is there a list of supported operators (with examples)?
Is it possible to produce a mongoDB query using `$in` operator?

**Comment by fredmaggiowski** [comment]
HI `$in` is supported since [rönd v1.7.0](https://github.com/rond-authz/rond/releases/tag/v1.7.0). 

Here you can find some examples: https://github.com/rond-authz/rond/blob/9b1a4dbf57dd24e38948ddfd3e8bc3fd0249501a/service/handler_test.go#L877

**Reply by SimoneBergonzi** [reply]
Hi,

I have the latest version of Rond,
![Screenshot from 2023-06-08 17-50-19](https://github.com/mia-platform/community/assets/26777944/199d9f7e-f044-4df8-8b69-b5ab24d98ebd)

but with this rule:
```
my_rows_filter {
  input.request.method == "GET"
  company := data.resources[_]
  managed_companies := input.user.properties.metadata.managed_companies
    
  managed_companies in company._id
}
```
I get the following error:
```
rego policies syntax is incorrect: 1 error occurred: Permissions:39: rego_parse_error: unexpected ident token: expected \n or ; or }
	managed_companies in company._id
	                  ^
```

**Reply by fredmaggiowski** [reply, marked as answer]
Where is this error from? it seems coming from the policy tester in Console, am i right?

Did you import the in keyword? https://www.openpolicyagent.org/docs/latest/policy-language/#futurekeywordsin

```rego
import future.keywords
```

**Reply by SimoneBergonzi** [reply]
Yep, importing `future.keywords` did the trick
Thanks!!
